HHS/simpler-grants-gov
# [Issue #3071] load New Relic browser code consistently

## Summary
Fixes #3071

### Time to review: __15 mins__

## Changes proposed
Refactors our layout situation to create a root layout which we can use to properly insert the client side New Relic code snippet into the head of the document consistently. This allows us to ensure that the snippet is generated on the server side at runtime, to fix bugs with consistency of loading the New Relic script that we have observed.

## Context for reviewers
### Static rendering

We should ensure that this does not break the build time rendering we have in place for the opportunity detail pages. @acouch, what is the best way to test that?

### Translation

Note that I was slightly concerned that changing the layout situation here would negatively impact our translation related code, but was able to confirm that it works fine with these changes. For more on testing that, you can see https://github.com/HHS/simpler-grants-gov/pull/3089

### Test steps
1. Update .env.production to point to your locally running API server
2. start a "production like" build locally with New Relic creds (see 1pass). `npm run build && NEW_RELIC_APP_NAME="<1pass>" NEW_RELIC_LICENSE_KEY="<1pass>" npm run start:nr`
3. visit all pages on the app and
4. _VERIFY_: page loads as expected
5. _VERIFY_: the <head> of the document contains a populated "nr-browser-agent" script. See screenshot
6. _VERIFY_: New Relic beacons are sent. Note that Firefox seems to block these beacons by default, so try testing in Chrome. Filter Network tab of dev tools for calls to https://bam.nr-data.net

## Additional information

![Screenshot 2024-12-04 at 9 19 48 AM](https://github.com/user-attachments/assets/baad8fbd-96bb-4ce5-a512-43f1775c231e)
